Update cyton_controller.py

saams4u opened the pull request:
Updated the controller class with the requisite actions.

saams4u committed:
Update cyton_controller.py

saams4u committed:
Update cyton_controller.py

deleomike commented on assistant/cyton/cyton_controller.py:
```diff
@@ -65,15 +68,71 @@ def set_pose(self, q: SE3):
+        robot_pickup_pose = self.robot.ikine_LM(item.pose, q0=self.pose)
+        traj = jtraj(self.pose, robot_pickup_pose, 100)
 
+        self.set_pose(traj)
```
I've been thinking about this a bit, let's make a function in the controller that runs trajectories. Something like

```python
def run_trajectory(self, qs: List[List]]):
    for q in qs:
         self.set_pose(q)
```

deleomike commented on assistant/cyton/cyton_controller.py:
```diff
@@ -65,15 +68,71 @@ def set_pose(self, q: SE3):
+        robot_dropoff_pose = self.robot.ikine_LM(location, q0=self.pose)
+        traj = jtraj(self.pose, robot_dropoff_pose, 100)
+
+        self.set_pose(traj)
```
Same case here, lets replace this with a run trajectory function call

deleomike commented on assistant/cyton/cyton_controller.py:
```diff
@@ -65,15 +68,71 @@ def set_pose(self, q: SE3):
+        Closes the gripper
+        :return: None
+        """
+        self.robot.close_gripper()
```
So for this I think we need to calculate the distance to close the gripper for. So this function should accept a block object as an input, and we should calculate the correct gripper distance to send based on the block dimensions. Mock pseudo code for this would be like

```python
def close_gripper(self, block):
    width = block.width
    self.set_gripper(width)
```

I think we'll have to play with this in the lab to make sure we're sending the right values for the robot.

deleomike commented on assistant/cyton/cyton_controller.py:
```diff
@@ -65,15 +68,71 @@ def set_pose(self, q: SE3):
+        Opens the gripper
+        :return: None
+        """
+        self.robot.open_gripper()
```
Similar to the other comment, lets move the logic for moving the gripper into a dummy function like

```python
def set_gripper(self, gripper_value):
    #dosomething
    pass
```

saams4u committed:
Update cyton_controller.py

Made the requisite changes to the controller class to include the `run_trajectory` and `set_gripper` functions with plausible `offset` values for the gripper.

saams4u reviewed (commented):
Made the requisite changes to the controller class to include the `run_trajectory` and the `set_gripper` functions with the `offset` values defined for the gripper.
